libs/common-services/src/main/java/uk/gov/di/ipv/core/library/helpers/DynamoDbHelper.java

```diff
@@ -0,0 +1,96 @@
+        Map<String, AttributeValue> out = new HashMap<>();
+        if (in != null) {
+            for (Map.Entry<String, Object> e : in.entrySet()) {
+                Object value = e.getValue();
+                if (value instanceof String valStr) {
+                    out.put(e.getKey(), AttributeValue.builder().s(valStr).build());
+                } else if (value instanceof Number valNbr) {
+                    out.put(e.getKey(), AttributeValue.builder().n(String.valueOf(valNbr)).build());
+                } else if (value instanceof LinkedHashMap) {
+                    marshallLinkedHashMapValue(out, e, (LinkedHashMap<String, String>) value);
+                } else {
+                    LOGGER.info(
+                            LogHelper.buildLogMessage(
+                                    "Not able to marshall it to lastEvaluateKey for scan."));
+                }
+            }
+        }
+        return out.isEmpty() ? null : out;
+    }
+
+    private static void marshallLinkedHashMapValue(
+            Map<String, AttributeValue> out,
+            Map.Entry<String, Object> e,
+            LinkedHashMap<String, String> value) {
+        var stringStringLinkedHashMap = value;
+        String[] aKeys =
+                stringStringLinkedHashMap
+                        .keySet()
+                        .toArray(new String[stringStringLinkedHashMap.size()]);
+        if (aKeys.length == 1) {
+            String linkedHashMapKeyValue = stringStringLinkedHashMap.get(aKeys[0]);
+            if (aKeys[0].equals("S")) {
+                out.put(e.getKey(), AttributeValue.builder().s(linkedHashMapKeyValue).build());
+            } else if (aKeys[0].equals("N")) {
+                out.put(e.getKey(), AttributeValue.builder().n(linkedHashMapKeyValue).build());
+            } else if (aKeys[0].equals("BOOL")) {
+                out.put(
+                        e.getKey(),
+                        AttributeValue.builder()
+                                .bool(Boolean.valueOf(linkedHashMapKeyValue))
+                                .build());
+            }
+        }
+    }
+
+    private static Object unwrapAttributeValue(AttributeValue av) {
+        if (av.nul() != null && av.nul()) return null;
+        if (av.m() != null && !av.m().isEmpty()) return unmarshallLastEvaluatedKey(av.m());
+        if (av.l() != null && !av.l().isEmpty())
+            return av.l().stream().map(DynamoDbHelper::unwrapAttributeValue).toList();
+        return Stream.<Function<AttributeValue, Object>>of(
+                        AttributeValue::s,
+                        AttributeValue::n,
+                        AttributeValue::bool,
+                        AttributeValue::ns,
+                        AttributeValue::ss,
+                        AttributeValue::b,
+                        AttributeValue::bs)
+                .map(f -> f.apply(av))
+                .filter(Objects::nonNull)
+                .findFirst()
+                .orElseThrow();
```

Comment:
Is this copied from somewhere? It's an interesting construction...

Reply:
yeah google.